apps/skilavottord/ws/src/app/modules/accessControl/accessControl.service.ts

```diff
@@ -1,69 +1,132 @@
 } from './accessControl.input'
+import { RecyclingPartnerModel } from '../recyclingPartner/recyclingPartner.model'
 
 @Injectable()
 export class AccessControlService {
-  constructor(@Inject(LOGGER_PROVIDER) private logger: Logger) {}
+  constructor(
+    @InjectModel(AccessControlModel)
+    private accessControlModel: typeof AccessControlModel,
+    @Inject(LOGGER_PROVIDER) private logger: Logger,
+  ) {}
 
   async findAll(): Promise<AccessControlModel[]> {
     this.logger.info('---- Starting findAll Access Controls ----')
+    try {
+      const res = await AccessControlModel.findAll({
+        include: [
+          {
+            model: RecyclingPartnerModel,
+          },
+        ],
+      })
+      this.logger.info(
+        'findAll-recyclingUsers result:' + JSON.stringify(res, null, 2),
+      )
+      return res
+    } catch (error) {
+      this.logger.error('error finding all recyclingUsers:' + error)
+    }
+  }
 
-    // TODO replace mock data with actual db query
-    const res = await Promise.resolve([
-      {
-        nationalId: '1234567777',
-        name: 'Gervimaður',
-        role: Role.recyclingCompany,
-        partnerId: '8888888888',
-      },
-      {
-        nationalId: '1234567888',
-        name: 'Gervimaður2',
-        role: Role.recyclingCompany,
-        partnerId: '9999999999',
-      },
-      {
-        nationalId: '1234567899',
-        name: 'Gervimaður3',
-        role: Role.recyclingCompany,
-        partnerId: '9999999999',
-      },
-    ] as AccessControlModel[])
-    return res
+  async findOne(nationalId: string): Promise<AccessControlModel> {
+    this.logger.info('find one access user...')
+    try {
+      const res = await this.accessControlModel.findOne({
+        where: { nationalId: nationalId },
+        include: [
+          {
+            model: RecyclingPartnerModel,
+          },
+        ],
+      })
+      this.logger.info(
+        'findOne-accessControl result:' + JSON.stringify(res, null, 2),
+      )
+      // null if not found
+      return res
+    } catch (error) {
+      this.logger.error('error finding one AccessControl:' + error)
+    }
   }
 
+  // async createAccess(
+  //   input: CreateAccessControlInput,
+  // ): Promise<AccessControlModel> {
+  //   // TODO replace mock data with actual db query
+  //   return Promise.resolve({
+  //     nationalId: '1234567890',
+  //     name: 'Gervimaður3',
+  //     role: Role.recyclingCompany,
+  //     partnerId: '9999999999',
+  //   })
+  // }
+
   async createAccess(
     input: CreateAccessControlInput,
   ): Promise<AccessControlModel> {
-    // TODO replace mock data with actual db query
-    return Promise.resolve({
-      nationalId: '1234567890',
-      name: 'Gervimaður3',
-      role: Role.recyclingCompany,
-      partnerId: '9999999999',
-    })
+    this.logger.info('Creating Access:' + JSON.stringify(input, null, 2))
+    try {
+      const dd = input as AccessControlModel
+      const res = await dd.save()
```

Comment:
This is a strange approach ... find it strange that it actually works... since you're just typescript asserting the object to `AccessControlModel` without actually creating the object, thus `save()` shouldn't exist.

Ditto on unnecessary redundant logs ... this could be:
```typescript
return this.accessControlModel.create({...input})
```